compiler/rustc_session/src/session.rs

```diff
@@ -590,7 +590,16 @@ impl Session {
     pub fn warn(&self, msg: impl Into<DiagnosticMessage>) {
         self.diagnostic().warn(msg)
     }
-    /// Delay a span_bug() call until abort_if_errors()
+
+    /// Call `span_bug` (ICE) with the provided span and message if no error has been emitted when `abort_if_errors` is called
+    /// (which is roughly after analysis and after codegen).
```

Comment:
My preference is to explain these things with a short introduction and a concise description of the behavior:

```suggestion
    /// Ensures that compilation cannot succeed.
    ///
    /// If this function has been called but no errors have been emitted and
    /// compilation succeeds, it will cause an internal compiler error (ICE).
```

Also, there are a *lot* of places where `abort_if_errors` is called. Is there any place in particular that is special?

Reply:
i dont think so, it's just after some amount of work has been completed. I guess one interesting case is the default `Value` impl to recover from query cycles, but that doesn't really matter here.